plugins/DuskVerb/tools/tuner/perceptual_diff.py

```
@@ -0,0 +1,132 @@
+THIRD = np.array([F0 * 2**(n/3) for n in range(-17, 13)])   # ~20Hz..16kHz
+# ISO 226-ish equal-loudness weight (relative dB, ear sensitivity ~ inverse of
+# threshold). Coarse: roll off lows hard, peak ~3-4kHz, gentle HF roll. Used to
+# weight the divergence RANK so a 3dB sub error doesn't outrank a 3dB presence error.
+def ear_weight(f):
+    # A-weighting-ish (dB), normalized to 0 at peak
+    f2 = f*f
+    ra = (12194.0**2 * f2*f2) / ((f2+20.6**2)*np.sqrt((f2+107.7**2)*(f2+737.9**2))*(f2+12194.0**2))
+    return 20*np.log10(ra+1e-12) + 2.0
+
+def onset(x, sr, drop_db=40.0):
+    e = np.abs(hilbert(x)); w = max(int(0.002*sr),1)
```

Ruff (0.15.18)

[error] Multiple statements on one line (semicolon)

(E702)